cdap-hbase-compat-1.2-cdh5.7.0/src/main/java/co/cask/data2/txprune/DefaultHBaseTransactionPruningPlugin.java

```diff
@@ -19,16 +19,43 @@
 import co.cask.cdap.data2.transaction.coprocessor.hbase12cdh570.DefaultTransactionProcessor;
 import co.cask.cdap.data2.transaction.messaging.coprocessor.hbase12cdh570.MessageTableRegionObserver;
 import co.cask.cdap.data2.transaction.queue.coprocessor.hbase12cdh570.HBaseQueueRegionObserver;
+import org.apache.hadoop.hbase.HColumnDescriptor;
 import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableExistsException;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.tephra.hbase.txprune.DataJanitorState;
 import org.apache.tephra.hbase.txprune.HBaseTransactionPruningPlugin;
 import org.apache.tephra.txprune.TransactionPruningPlugin;
 
+import java.io.IOException;
+
 /**
  * {@link TransactionPruningPlugin} for CDAP Datasets, TMS Message Table, HBase Queues.
  */
 public class DefaultHBaseTransactionPruningPlugin extends HBaseTransactionPruningPlugin {
 
   @Override
+  protected void createPruneTable(TableName stateTable) throws IOException {
```

Comment:
This is because there is a API difference between hbase-1.2-cdh5.7.0. HTableDescriptor#addFamily returns null in this version while it returns HTableDescriptor in the HBase 1.1 tephra compat which is what this version depends on.